### ISSUE TITLE
[Bug]: Apps outlive installations

### ISSUE THREAD

**Issue by eeejay** [flatpak/flatpak#5775]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

1.15.6

### What Linux distribution are you using?

Fedora Linux

### Linux distribution version

39

### What architecture are you using?

x86_64

### How to reproduce

1. Spawn a D-Bus service exported by a flatpak
2. Uninstall the flatpak

### Expected Behavior

Service should be shut down.

### Actual Behavior

Service is still running

### Additional Information

This happens with apps too, but is more acute with services since the user does not have visibility into the fact that a stale service is still running. This introduces also safety concerns since the user believes that they removed a malicious flatpak only to still have it running in the background.

There is also the problem that a user might upgrade a flatpak with an exported service, but the older one is still running until the user exits the session.

I propose that services either be forcibly killed, or that they expose a special interface with a `Shutdown()` method in order to do any cleanup. This method should make the service quit in-line so the method will ultimately "fail".

The first solution is good for legacy apps, the second solution could be opted in to if needed.

**Comment by hydra** [comment]
I fully agree with this, as I detailed in my similar bug report.  Addtionally the server could be reading/writing/creating data which the user may have opted to delete as part of the flatpak removal.

https://github.com/flatpak/flatpak/issues/6459
